### PR TITLE
Adds some sanity to lighting_source.dm

### PR DESCRIPTION
**Pull request by MadmanMartian**
[04:55:53] Runtime in lighting_source.dm,281: type mismatch: null -= /datum/light_source (/datum/light_source) seems to imply that as it itterates through the turfs at that point where it tries to remove itself from that turfs affecting_lights, it finds that it isn't in the turfs affecting lights, and crashes, causing light to cease functioning.

A little bit of sanity to make sure the turfs affecting lights both exists, and contains the src, should help resolve this issue

### CODE REVIEW

File: code/modules/lighting/lighting_source.dm [hunk 1]
```diff
@@ -278,7 +278,8 @@
 	L = affecting_turfs - turfs // Now-gone turfs, remove us from the affecting lights.
 	affecting_turfs -= L
 	for (var/turf/T in L)
-		T.affecting_lights -= src
+		if(T.affecting_lights && T.affecting_lights.Find(src))
```
Comment (unid15) [inline code/modules/lighting/lighting_source.dm]: list.Remove(A) works 100% fine if list doesn't contain A in the first place, so the second check is not needed here atm

**Comment by MadmanMartian** [comment]
@DamianX I'd done what you had asked, but pushed it to the wrong branch.

Whoops

**Comment by DamianX** [comment]
To be clear

- it wasn't me
- `-=` is exactly the same as `.Remove`

**Comment by MadmanMartian** [comment]
further whoops

**Comment by PJB3005** [comment]
I was looking at the changes to the lighting engine and stumbled across this.

I'm PRETTY sure this is just masking the error that turf.affecting_lights flat out doesn't exist. This should't be possible.

**Comment by DamianX** [comment]
wouldn't be the first place in which that's checked
see this for instance
https://github.com/vgstation-coders/vgstation13/blob/e822b7cd466b80aa862a2c668223c08387fc6dce/code/modules/lighting/lighting_source.dm#L273-L281

**Comment by PJB3005** [comment]
That's completely different. That's the initialization of the list of the
first place (the list is lazily initialized).

That code should ALWAYS have ran on the turf before the check added in this
PR. That's the problem. The list ALWAYS has to exist.

**Comment by PJB3005** [comment]
This PR is just silencing a runtime instead of actually fixing it. IMO it should be reverted and let the runtime just be spammed again as a reminder to fix it properly.

While unlikely, this can be a real point of confusion if somebody else has to later fix the root cause of this.

**Pull request merged**